stac_fastapi/elasticsearch/stac_fastapi/elasticsearch/core.py

```diff
@@ -357,28 +363,30 @@ def create_collection(
             collection_id=collection["id"], base_url=base_url
         ).create_links()
         collection["links"] = collection_links
-        self.database.create_collection(collection=collection)
+        await self.database.create_collection(collection=collection)
 
         return CollectionSerializer.db_to_stac(collection, base_url)
 
     @overrides
-    def update_collection(
+    async def update_collection(
         self, collection: stac_types.Collection, **kwargs
     ) -> stac_types.Collection:
         """Update collection."""
         base_url = str(kwargs["request"].base_url)
 
-        self.database.find_collection(collection_id=collection["id"])
-        self.delete_collection(collection["id"])
-        self.create_collection(collection, **kwargs)
+        await self.database.find_collection(collection_id=collection["id"])
+        await self.delete_collection(collection["id"])
+        await self.create_collection(collection, **kwargs)
 
         return CollectionSerializer.db_to_stac(collection, base_url)
 
     @overrides
-    def delete_collection(self, collection_id: str, **kwargs) -> stac_types.Collection:
+    async def delete_collection(
+        self, collection_id: str, **kwargs
+    ) -> stac_types.Collection:
         """Delete collection."""
-        self.database.delete_collection(collection_id=collection_id)
-        return None
+        await self.database.delete_collection(collection_id=collection_id)
+        return None  # type: ignore
```

Comment:
Is this worth PR-ing onto stac-fastapi to get the signature of `delete_collection` changed to return `None`?

Reply:
I filed this issue the other day about this https://github.com/stac-utils/stac-fastapi/issues/376

Ideally, they'd all be Optional, and if they returned None, then a 204 No Content would be the response rather than just 200 and whatever it sends now.